lib/modules/apostrophe-login/index.js

```diff
@@ -315,22 +368,52 @@ module.exports = {
             if (!user) {
               throw 'notfound';
             }
-            // Gets i18n'd in the template, also bc with what templates that tried to work
-            // before certain fixes would expect (this is why we still pass a string and not
-            // a flag, and why we call it `message`)
-            return self.sendPage(req, 'passwordReset', { reset: reset, email: email });
+            return self.sendPage(req, 'passwordReset', { reset: reset, email: email, message: getMessage(), errors: req.query.errors });
           }).catch(function(err) {
             self.apos.utils.error(err);
             return res.redirect('/login?message=' + req.__('That reset code was not found. It may have expired. Try resetting again.'));
           });
+
+          function getMessage() {
+            let message;
+            if (req.query.message) {
+              message = req.query.message;
+            }
+            if (req.query.errors && req.query.errors.length) {
+              message = req.query.errors.join(', ');
+            }
+            if (message === undefined) {
+              return undefined;
+            }
+            return self.apos.utils.capitalizeFirst(message);
+          }
         });
 
         self.apos.app.post('/password-reset', function(req, res) {
           var reset = self.apos.launder.string(req.body.reset);
           var email = self.apos.launder.string(req.body.email);
           var password = self.apos.launder.string(req.body.password);
-          if ((!reset.length) || (!password.length)) {
-            return res.redirect('/password-reset-request?error=missing');
+          if (!reset.length) {
+            return res.redirect('/password-reset?' + qs.stringify({
+              errors: [ 'The password reset code is missing from your link. Try resetting your password again.' ],
```

Comment:
Suggestion: "The password reset code is missing from your URL. Click the link in your reset email or try resetting your password again."